base_layer/wallet/src/storage/database.rs

```diff
@@ -351,6 +359,41 @@ where T: WalletBackend + 'static
         self.db.clear_scanned_blocks_before_height(height, exclude_recovered)?;
         Ok(())
     }
+
+    // ----------------------------------------------------------------------------
+    // burnt proofs
+
+    pub fn insert_burnt_proof(
+        &self,
+        id: i32,
+        reciprocal_claim_public_key: String,
+        proof: String,
+        burned_at: NaiveDateTime,
+    ) -> Result<(), WalletStorageError> {
+        self.db.write(WriteOperation::Insert(DbKeyValuePair::BurntProof((
```

Comment:
I know the current implementation uses this pattern a lot, but I actually don't like it and I'd prefer to not increase the tech debt. Rather implement the methods individually.

Look at `self.db.clear_scanned_blocks` for the preferred pattern

Reply:
Same goes for all the other methods you've implemented in this file

Reply:
yeah, having an explicit trait interface that does exactly what you want and takes in exactly what you want without some wrapper is much cleaner, we have been removing the this `DbKeyValuePair` from the codebase as we add more and more code

Reply:
done